File: getting_started/javascript-client/update_data.js
const TerminusClient = require("@terminusdb/terminusdb-client");

// Connecting to TerminusX
// TODO: Change teamname
const client = new TerminusClient.WOQLClient(
  "https://cloud.terminusdb.com/cloudux/",
  { user: "username", organization: "cloudux", db: "GettingStartedDB" }
);

//Assign your key to environment variable TERMINUSDB_ACCESS_TOKEN
client.setApiKey(process.env.TERMINUSDB_ACCESS_TOKEN);

const updateAndLinkData = async () => {
    const destiny = await client.getDocument({"id":"Employee/001"});

    // have to delete "@id" because database will create a new one
    delete destiny.address['@id'];

    destiny.address.postcode = "PH12 3RP";
    destiny.address.street = "Lairg Road";
    destiny.address.street_num = 73;
    destiny.address.town = "Newbigging";

    await client.updateDocument(destiny,{},"","updating 001");

    const ethan = {
        "@type": "Employee",
        "employee_id": "005",
        name: "Ethan Abbott",
        title: "Backend Developer",
        team: "IT",
        contact_number: "070 7796 8035",
        address: {
            "@type": "Address",
            postcode: "IV27 2TG",
            street: "Shore Street",
            street_num: 84,
            town: "Stoer"
        },
        manager: "Employee/004",
    }
    await client.addDocument(ethan,{},"","Adding ethan");

    const result = await client.getDocument({"as_list":true});
    console.log(result);
    
}

updateAndLinkData();

File: getting_started/python-client/update_data.py
from terminusdb_client import Client
from terminusdb_client.woqlschema import WOQLSchema

# For Terminus X, use the following
# client = Client("https://cloud.terminusdb.com/<Your Team>/")
# client.connect(db="demo_workshop", team="<Your Team>", use_token=True)

client = Client("http://127.0.0.1:6363/")
client.connect(db="getting_started")

data_schema = WOQLSchema()
data_schema.from_db(client)

# Update a document

destiny_raw = client.get_document("Employee/001")
destiny = data_schema.import_objects(destiny_raw)

destiny.address.postcode = "PH12 3RP"
destiny.address.street = "Lairg Road"
destiny.address.street_num = 73
destiny.address.town = "Newbigging"

client.update_document(destiny, commit_msg="Update Destiny")

# Linking a new document to an old document

Employee = data_schema.object.get("Employee")
Address = data_schema.object.get("Address")
Team = data_schema.object.get("Team")

ethan_address = Address(
    postcode="IV27 2TG", street="Shore Street", street_num=84, town="Stoer"
)

manager_raw = client.get_document("Employee/004")
ethan_manager = data_schema.import_objects(manager_raw)

ethan = Employee(
    _id="Employee/005",
    name="Ethan Abbott",
    title="Backend Developer",
    team=Team.it,
    contact_number="070 7796 8035",
    address=ethan_address,
    manager=ethan_manager,
)

client.update_document(ethan, commit_msg="Adding Ethan")
